quic/s2n-quic-tls/Cargo.toml

```diff
@@ -20,13 +21,15 @@ libc = "0.2"
 s2n-codec = { version = "=0.3.0", path = "../../common/s2n-codec", default-features = false }
 s2n-quic-core = { version = "=0.17.0", path = "../s2n-quic-core", default-features = false }
 s2n-quic-crypto = { version = "=0.17.0", path = "../s2n-quic-crypto", default-features = false }
-s2n-tls = { version = "=0.0.21", features = ["quic"] }
+s2n-tls = { version = "=0.0.26", features = ["quic"] }
 
 [target.'cfg(all(s2n_quic_unstable, s2n_quic_enable_pq_tls))'.dependencies]
-s2n-tls = { version = "=0.0.21", features = ["quic", "pq"] }
+s2n-tls = { version = "=0.0.26", features = ["quic", "pq"] }
 
 [dev-dependencies]
 checkers = "0.6"
+pin-project = { version = "1" }
+openssl = { version = "0.10" }
 # Build the vendored version to make it easy to test in dev
```

Comment:
I don't love adding this dependency, but I'm not sure how else to test a successful handshake. I could test an unsuccessful handshake without it, and maybe that'd be sufficient to at least prove that the callback was triggered?

If we really don't want to add the dependency and also really want a successful test, I could go back and add (s2n_async_pkey_op_perform)[https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L2886] to the s2n-tls Rust bindings and use that in the s2n-quic test.

Reply:
> add [s2n_async_pkey_op_perform](https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L2886) to the s2n-tls Rust bindings and use that in the s2n-quic test.

Since the whole point if offloading is to allow not relying on s2n-tls, using an external crypto lib as the provider for test does seem to make sense, as long as we're not worried about test stability.

It's also a great form of implicit differential testing...

Reply:
Note that we already pull in `openssl-sys` for `s2n-tls-sys` so it's not a big deal. And dev-dependencies are less of an issue, anyway.